libs/sdk-ui-ext/src/internal/interfaces/Visualization.ts

```diff
@@ -32,12 +32,16 @@ import { IAvailableSortsGroup, ISortConfig } from "./SortConfig";
 
 export type RenderFunction = (component: any, target: Element) => void;
 
+export type ElementSelectorFunction = () => HTMLElement | null;
+/** @deprecated Use ElementSelectorFunction instead */
+export type ElementSelectorString = string;
+
 export interface IVisConstruct {
     backend: IAnalyticalBackend;
     // TODO: rename to workspace
     projectId: string;
-    element: string;
-    configPanelElement: string;
+    element: ElementSelectorString | ElementSelectorFunction;
+    configPanelElement: ElementSelectorString | ElementSelectorFunction;
```

Comment:
It seems that AD does not use this API directly, so technically we could drop support for `ElementSelectorString` anytime (if we account only for internal users).

Reply:
If that is the case, then yes, this should not be used by anyone else.